ntbs-service/Pages/Notifications/Edit/Table/ManualTestResult.cshtml.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using ntbs_service.DataAccess;
+using ntbs_service.Helpers;
+using ntbs_service.Models;
+using ntbs_service.Models.Enums;
+using ntbs_service.Services;
+
+namespace ntbs_service.Pages.Notifications.Edit
+{
+    public class ManualTestResultPage : NotificationEditModelBase
+    {
+        private readonly IReferenceDataRepository _referenceDataRepository;
+        private readonly ITestResultsRepository _testResultsRepository;
+
+        public SelectList ManualTestTypes { get; set; }
+        public SelectList SampleTypes { get; set; }
+        public SelectList ResultOptions { get; set; }
+
+        [BindProperty(SupportsGet = true)]
+        public int? RowId { get; set; }
+        [BindProperty]
+        public ManualTestResult TestResultForEdit { get; set; }
+        [BindProperty]
+        public FormattedDate FormattedTestDate { get; set; }
+
+        public ManualTestResultPage(
+            INotificationService notificationService,
+            IAuthorizationService authorizationService,
+            INotificationRepository notificationRepository,
+            IReferenceDataRepository referenceDataRepository,
+            ITestResultsRepository testResultsRepository)
+            : base(notificationService, authorizationService, notificationRepository)
+        {
+            _referenceDataRepository = referenceDataRepository;
+            _testResultsRepository = testResultsRepository;
+        }
+
+        protected override async Task ValidateAndSave()
+        {
+            TestResultForEdit.NotificationId = NotificationId;
+            TestResultForEdit.Dob = Notification.PatientDetails.Dob;
+            await SetRelatedEntities();
+            SetDate();
+
+            if (TryValidateModel(TestResultForEdit, "TestResultForEdit"))
+            {
+                if (RowId != null)
+                {
+                    await _testResultsRepository.AddTestResultAsync(TestResultForEdit);
+                }
+                else
+                {
+                    await _testResultsRepository.UpdateTestResultAsync(TestResultForEdit);
+                }
+            }
+        }
+
+        protected override async Task<IActionResult> PrepareAndDisplayPageAsync(bool isBeingSubmitted)
+        {
+            if (RowId != null)
+            {
+                TestResultForEdit = Notification.TestData.ManualTestResults
```

Comment:
Can't we just fetch this directly from context.ManualTestResult.Find(RowId), rather than going via notification?

Reply:
Yeah could do - that actually feels better as we are then not requiring the Notification fetch to necessarily join onthe relevant info!

Reply:
ah actually, we would then have to manually check that the test belongs to the notification, which we get for free here - so i think i'll leave it as is.